sklearn/impute/_base.py

```diff
@@ -72,12 +72,22 @@ def __init__(self, missing_values=np.nan, add_indicator=False):
         self.missing_values = missing_values
         self.add_indicator = add_indicator
 
+    def _change_missing_values(self, new_missing_values):
+        """Add a new attribute for missing values."""
+        if not hasattr(self, '_missing_values'):
+            self._missing_values = new_missing_values
+
     def _fit_indicator(self, X):
```

Comment:
This `self._missing_values` logic is very obfuscating. Why don't you just have a `missing_values` param here? Indeed, it looks like you change every use of `_fit_indicator` in this PR, such that `_fit_indicator(mask)` should suffice.

Reply:
Yup, I also did not feel good while doing these changes but I wasn't sure that I should play with `_fit_indicator` function, and hence ended up making new one. The other approach was of passing `_fit_indicator` a default parameter of `missing_mask`, I'll switch to this approach in the next commit.